### PR TITLE
Add service protocol method to facilitate getting snapshots

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

File: shell/common/shell_unittests.cc [hunk 1]
```diff
@@ -2524,6 +2524,41 @@ TEST_F(ShellTest, OnServiceProtocolEstimateRasterCacheMemoryWorks) {
...
+  ServiceProtocol::Handler::ServiceProtocolMap empty_params;
+  rapidjson::Document document;
+  OnServiceProtocol(
+      shell.get(), ServiceProtocolEnum::kRenderFrameWithRasterStats,
+      shell->GetTaskRunners().GetRasterTaskRunner(), empty_params, &document);
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  document.Accept(writer);
+  std::string expected_json =
+      "\"snapshot\":[137,80,78,71,13,10,26,10,0,"
```
Comment (dnfield) [inline shell/common/shell_unittests.cc:2550]: consider adding a doc comment explaining how this was generated and what might cause it to change (and what an acceptable change would be).

File: shell/common/shell_unittests.cc [hunk 2]
```diff
@@ -2524,6 +2524,41 @@ TEST_F(ShellTest, OnServiceProtocolEstimateRasterCacheMemoryWorks) {
...
+      "0,0,13,73,72,68,82,0,0,0,1,0,0,0,1,8,6,0,0,0,31,21,196,137,0,0,0,1,115,"
+      "82,71,66,0,174,206,28,233,0,0,0,4,115,66,73,84,8,8,8,8,124,8,100,136,0,"
+      "0,0,13,73,68,65,84,8,153,99,248,207,192,240,31,0,5,0,1,255,171,206,54,"
+      "137,0,0,0,0,73,69,78,68,174,66,96,130]";
+  std::string actual_json = buffer.GetString();
+  EXPECT_THAT(actual_json, ::testing::HasSubstr(expected_json));
```
Comment (dnfield) [inline shell/common/shell_unittests.cc:2556]: Are there JSON methods we could be using instead to validate fields? e.g. validate that snapshot is a list of integers, and validate that the timing field is present?

Comment (iskakaushik) [reply shell/common/shell_unittests.cc:2556]: I could maybe parse it out with `rapidjson` and validate that the timing field exists. I was being lazy and using `HasSubstring`. Let me change it to do real parsing instead.

Comment (dnfield) [reply shell/common/shell_unittests.cc:2556]: (I'm mainly worried that this test will break if we move to a different rendering backend or the rendering backend slightly changes things and it'll take more work to validate)

**Review by dnfield — APPROVED** [review]
Change LG, if you can improve the test it would be nice (or at the least please add a comment about how to update it and what it should visually look like)

**Comment by iskakaushik** [comment]
I improved the test and added a comment on how to make it even better.

**Pull request merged**